### PR TITLE
Add more known Regex patterns to increase test coverage.

### PR DESCRIPTION
**Pull request by eerhardt**
Adding a few real-world Regex patterns I found.

### CODE REVIEW

**Review by stephentoub — APPROVED** [review]
Thanks for adding these!

**Comment by danmosemsft** [comment]
https://github.com/dotnet/corefx/issues/27878

File: src/libraries/System.Text.RegularExpressions/tests/Regex.KnownPattern.Tests.cs [hunk 1]
```diff
@@ -136,5 +136,88 @@ string DomainMapper(Match match)
...
+        [InlineData("_IsValidCSharpName", true)]
+        [InlineData("__", true)]
+        [InlineData("a\u2169", true)] // \u2169 is in {Nl}
+        [InlineData("\u2169b", true)] // \u2169 is in {Nl}
+        [InlineData("a\u0600", true)] // \u0600 is in {Cf}
+        [InlineData("\u0600b", false)] // \u0600 is in {Cf}
+        [InlineData("a\u0300", true)] // \u0300 is in {Mn}
+        [InlineData("\u0300b", false)] // \u0300 is in {Mn}
+        [InlineData("https://foo.com:443/bar/17/groups/0ad1/providers/Network/public/4e-ip?version=16", false)]
+        [InlineData("david.jones@proseware.com", false)]
```
Comment (danmosemsft) [inline src/libraries/System.Text.RegularExpressions/tests/Regex.KnownPattern.Tests.cs]: Nit: I see this came from some ancient BCL blog (at least) but I suggest to replace with a foo@contoso.com address that definitely doesn't belong to anyone.

**Review by danmoseley — APPROVED** [review]

File: src/libraries/System.Text.RegularExpressions/tests/Regex.KnownPattern.Tests.cs (hunk 1 above)
Comment (stephentoub) [reply src/libraries/System.Text.RegularExpressions/tests/Regex.KnownPattern.Tests.cs]: > I see this came from some ancient BCL blog

It's actually in the docs:
https://docs.microsoft.com/en-us/dotnet/standard/base-types/how-to-verify-that-strings-are-in-valid-email-format#compile-the-code

I just put all of the docs samples into our tests, basically verbatim.

Comment (eerhardt) [reply src/libraries/System.Text.RegularExpressions/tests/Regex.KnownPattern.Tests.cs]: And I just copied it from the above test.

I can change all the emails to `foo.bar@contoso.com` in this file, if you two think it is necessary.

Comment (eerhardt) [reply src/libraries/System.Text.RegularExpressions/tests/Regex.KnownPattern.Tests.cs]: Funny enough, `proseware.com` redirects me to `https://www.microsoft.com/en-us/`. So I think it is safe. 😉

**Pull request merged**